stats/identificator.go

```diff
@@ -0,0 +1,117 @@
+package stats
+
+import (
+	"go.uber.org/zap"
+	"strings"
+)
+
+type metric struct {
+	ExtractedMetric string
+	ApplicationName string
+	ApplicationType ApplicationType
+	IsPrometheus    bool
+}
+
+type ApplicationType string
+
+const (
+	Agglo         ApplicationType = "agglo"
+	Aggregated    ApplicationType = "aggregated"
+	AggregatedAll ApplicationType = "aggregated-all"
+	StartCriteo   ApplicationType = "start-criteo"
+	LegacyHosting ApplicationType = "legacy-hosting"
+	StartAppName  ApplicationType = "start-appname"
+	None          ApplicationType = "None"
+)
+
+func GetMetric(logger *zap.Logger, metricPath string, componentLen int) metric {
+	if componentLen < 1 {
+		logger.Panic("components should be > 0", zap.Int("componentLen", componentLen))
+	}
+	components := getComponents(metricPath, componentLen)
+	applicationType := getApplicationType(components)
+	applicationName := getApplicationName(components, applicationType)
+	extractedMetric := strings.Join(components, ".")
+	isPrometheus := isPrometheusMetric(components)
+	return metric{ExtractedMetric: extractedMetric, ApplicationName: applicationName, ApplicationType: applicationType, IsPrometheus: isPrometheus}
+}
+
+func getComponents(metricPath string, componentsLen int) []string {
+	currentIndex := 0
+	components := make([]string, componentsLen)
+	for componentNb := 0; componentNb < componentsLen; componentNb++ {
+		nextDotIndex := strings.IndexByte(metricPath[currentIndex:], '.')
+		if nextDotIndex != -1 {
+			components[componentNb] = metricPath[currentIndex : currentIndex+nextDotIndex]
+		} else {
+			components[componentNb] = metricPath[currentIndex:]
+			components = components[:componentNb+1]
+			break
+		}
+		currentIndex += nextDotIndex + 1
+	}
+	return components
+}
+func getApplicationType(components []string) ApplicationType {
```

Comment:
This is very specific to criteo, it would make more sense to have this classification configurable via a config file instead